src/transports/imap/imap_transport.php

```diff
@@ -563,6 +601,14 @@ public function authenticate( $user, $password )
             $this->state = self::STATE_NOT_CONNECTED;
             return false;
         }
+        // If we are using the XOAuth2 SASL mechanism we will get a base64 encoded response back to get standard
+        // response back we then need to send a blank response "\r\n" to the authentication response.
```

Comment:
This only needs to happen when there was an [error](https://developers.google.com/gmail/imap/xoauth2-protocol#error_response_2)? The [success](https://developers.google.com/gmail/imap/xoauth2-protocol) looks not to need this. So in that case, should it always send this new line?